money.go

```diff
@@ -203,13 +234,13 @@ func (m *Money) Allocate(rs []int) ([]*Money, error) {
 
 	// Calculate leftover value and divide to first parties
 	lo := m.amount.val - total
-	sub := 1
+	sub := int64(1)
 	if lo < 0 {
 		sub = -1
```

Comment:
Correct. It is alright. `-1` is an untyped integer, which is automatically converted to `int64`.